### PR TITLE
clean up unstructured conversion

### PR DESCRIPTION
**Pull request by Garrybest**
Signed-off-by: Garrybest <garrybest@foxmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controllers/status/cluster_status_controller.go [hunk 1]
```diff
@@ -455,8 +456,8 @@ func convertObjectsToNodes(nodeList []runtime.Object) ([]*corev1.Node, error) {
 	nodes := make([]*corev1.Node, 0, len(nodeList))
 	for _, obj := range nodeList {
 		unstructObj := obj.(*unstructured.Unstructured)
-		node := &corev1.Node{}
-		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructObj.UnstructuredContent(), node); err != nil {
+		node, err := helper.ConvertToNode(unstructObj)
```
Comment (XiShanYongYe-Chang) [inline pkg/controllers/status/cluster_status_controller.go:459]: This local variable `unstructObj` can also be removed, directly use `obj.(*unstructured.Unstructured)`.

**Comment by RainbowMango** [comment]
/lgtm
/approve

/hold
Feel free to un-hold after all checks succeed.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Could you please modify these two places together?

https://github.com/karmada-io/karmada/blob/c4e22717dfff7e87301236cf218df7a636bc2164/pkg/controllers/mcs/endpointslice_controller.go#L92

https://github.com/karmada-io/karmada/blob/c4e22717dfff7e87301236cf218df7a636bc2164/pkg/webhook/work/mutating.go#L124

File: pkg/controllers/status/cluster_status_controller.go (hunk 1 above)
Comment (Garrybest) [reply pkg/controllers/status/cluster_status_controller.go:459]: OK

Comment (XiShanYongYe-Chang) [reply pkg/controllers/status/cluster_status_controller.go]: Thank you very much.

**Pull request merged**